tests/modeltests/tst_callgraphgenerator.cpp

```diff
@@ -0,0 +1,139 @@
+#include "callgraphgenerator.h"
+
+#include <QDebug>
+#include <QObject>
+#include <QSignalSpy>
+#include <QTemporaryFile>
+#include <QTest>
+
+#include "../../src/parsers/perf/perfparser.h"
+#include "../../src/perfrecord.h"
+#include "../testutils.h"
+#include "data.h"
+
+QString findExe(const QString& name)
+{
+    QFileInfo exe(QCoreApplication::applicationDirPath() + QLatin1String("/../test-clients/%1/%1").arg(name));
+    qDebug() << exe;
+    VERIFY_OR_THROW(exe.exists() && exe.isExecutable());
+    return exe.canonicalFilePath();
+}
+
+class TestCallgraphGenerator : public QObject
+{
+    Q_OBJECT
+private slots:
+    void initTestCase()
+    {
+        qSetGlobalQHashSeed(0);
```

Comment:
why? I'd much rather not do this in a test to prevent us from not seeing the random effects of unordered containers